File: degwer/normal/630/P.cpp
#include<stdio.h>
#include<math.h>
double pi=3.1415926535897932384626433832;
int main()
{
    double p,r;
    scanf("%lf%lf",&p,&r);
    printf("%.10lf\n",r*r*p*sin(pi/(p*2))*(sin(pi/p)/sin(pi*3/(p*2))));
}

File: lewin/normal/630/P.py
import sys
import math

f = sys.stdin
n,r = map(int, f.readline().split())

side = r
ang1 = math.pi / n
ang2 = math.pi / 2.0 / n
ang3 = math.pi - ang1 - ang2
subarea = side * side * math.sin(ang1) * math.sin(ang2) / math.sin(ang3)

print subarea * n